octo/model/octo_model.py

```diff
@@ -175,29 +187,46 @@ def sample_actions(
         Args:
             observations: dictionary of arrays of shape (batch_size, window_size, *)
             tasks: dict of tasks of shape (batch_size, *)
-            pad_mask: (batch_size, window_size) Boolean mask that is False when the timestep corresponds to padding
+            unnormalization_statistics: dict of statistics for unnormalizing actions (must contain "mean",
+                "std", and optionally "mask")
+            timestep_pad_mask: (batch_size, window_size) Boolean mask that is False when the timestep corresponds to padding
             train: whether to run in train mode
             ...see `action_heads.py` for the rest of the kwargs.
         Returns:
-            actions: (*sample_shape, batch_size, pred_horizon, action_dim)
+            actions: (*sample_shape, batch_size, action_horizon, action_dim)
         """
-        if pad_mask is None:
-            pad_mask = observations["pad_mask"]
+        if timestep_pad_mask is None:
+            timestep_pad_mask = observations["timestep_pad_mask"]
 
         transformer_outputs = self.run_transformer(
-            observations, tasks, pad_mask, train=train
+            observations, tasks, timestep_pad_mask, train=train
         )
         action_head: ActionHead = self.module.bind({"params": self.params}).heads[
             "action"
         ]
-        return action_head.predict_action(
+        action = action_head.predict_action(
             transformer_outputs,
             train=train,
             argmax=argmax,
             sample_shape=sample_shape,
             rng=rng,
             temperature=temperature,
+            embodiment_action_dim=len(unnormalization_statistics["mean"])
+            if unnormalization_statistics is not None
+            else None,
         )
+        if unnormalization_statistics is not None:
```

Comment:
Did we completely drop support for bounds normalization? Could there be an issue here where they train on bounds normalization, but accidentally evaluate with normal unnormalization

Reply:
Good catch, I updated sample_actions to take in a normalization type.